client/src/partials/patient_edit/edit_patient.js

```diff
@@ -0,0 +1,269 @@
+          return true;
+        }
+
+        // Sensible year limits - may need to change to accomodate legacy patients
+        if (year > session.current_year || year < 1900) {
+	  // TODO: Maybe not necessary any more since the form/angular validates Date on the fly ???
+          validation.dates.flag = validation.dates.tests.limit;
+          return true;
+        }
+      }
+      return false;
+    }
+
+    // Tests in an ng-disabled method often got called in the wrong order/ scope was not updated
+    $scope.$watch('patient.dob', function (nval, oval) {
+      customValidation();
+    }, true);
+
+    $scope.$watch('session', function (nval, oval) {
+      customValidation();
+    }, true);
+
+    $scope.changeDebitor = function () {
+      // session.mode = 'change_debitor';
+      // TODO Needs to be implemented!
+      alert('Not implemented yet!');
+    };
+
+    $scope.updatePatient = function () {
+      var patient = connect.clean(angular.copy($scope.patient));
+
+      // Update the locations
+      patient.origin_location_id = session.originLocationUuid;
+      patient.current_location_id = session.currentLocationUuid;
+
+      // Make sure the DOB is in SQL format
+      patient.dob = util.sqlDate(patient.dob);
+
+      // Normalize the patient names (just in case they have been changed)
+      patient.first_name = util.normalizeName(patient.first_name);
+      patient.last_name = util.normalizeName(patient.last_name);
+      patient.father_name = util.normalizeName(patient.father_name);
+      patient.mother_name = util.normalizeName(patient.mother_name);
+      patient.spouse = util.normalizeName(patient.spouse);
+      
+      // Get rid of any extraneous fields
+      delete patient.reference;
+      delete patient.registration_date;
+      delete patient.debitor_group_id;
+      delete patient.debitor_name;
+      delete patient.debitor_group_name;
+      delete patient.project_abbr;
+      delete patient.hr_id;
+
+      // Enable blank overwrites
+      editable_fields.forEach(function (fname) {
+	if (original_patient_data[fname] && !(patient[fname])) {
+	  patient[fname] = null;
+	}
+      });
+
+      // Save the patient data
+      connect.put('patient', [patient], ['uuid'])
+	.then(function () {
+	  messenger.success($translate.instant('PATIENT_EDIT.UPDATE_SUCCESS'));
+	})
+	.catch(function (err) {
+	  messenger.danger($translate.instant('PATIENT_EDIT.UPDATE_FAILED'));
+	  console.log(err);
+	});
+
+      session.mode = 'edit';
+    };
+
+    $scope.initialiseEditing = function initialiseEditing (selectedPatient) {
+      if (selectedPatient && 'uuid' in selectedPatient && selectedPatient.uuid) {
+        patient_uuid = $scope.patient_uuid = selectedPatient.uuid;
+	dependencies.patient.query.where[0] = 'patient.uuid=' + patient_uuid;
+	validate.process(dependencies)
+	  .then(startup);
+	session.mode = 'edit';
+      }
+      else {
+	messenger.danger($translate('ERR_MISSING'));
```

Comment:
Sounds like a discussion we should all have as a team.  We can add that discussion to the agenda.